beacon_node/src/cli.rs

```diff
@@ -1,28 +1,40 @@
-use clap::{App, Arg};
+use clap::Arg;
+use clap_utils::{flags::*, DefaultConfigApp as App};
+use std::collections::HashMap;
 
-pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
-    App::new("beacon_node")
+pub fn cli_app<'a>(file_args: Option<&'a HashMap<&'a str, &'a str>>) -> Result<App<'a>, String> {
+    if let Some(args) = file_args {
+        for key in args.keys() {
+            if !BEACON_NODE_FLAGS.contains(key)
+                && !BEACON_BOOT_NODE_FLAGS.contains(key)
+                && !BEACON_VALIDATOR_FLAGS.contains(key)
+                && !GLOBAL_FLAGS.contains(key)
+            {
+                return Err(format!("--{} is not a valid beacon node flag.", key));
+            }
+        }
+    }
+
+    Ok(App::new("beacon_node", file_args)
         .visible_aliases(&["b", "bn", "beacon"])
-        .version(crate_version!())
```

Comment:
This is no longer available in `clap` do we want an alternative? I didn't think crate-level versions seemed really relevant